### PR TITLE
[Fix] Fix the wrong index of coor in middle_encoder.pillar_scatter.forward_single

### PR DESCRIPTION
**Pull request by hongye-dev**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
something wrong when i use the function middle_encoder.pillar_scatter.forward_single

## Modification
change the index of coors in line 53

### CODE REVIEW

**Comment by hongye-dev** [comment]
> [CLA assistant check](https://cla-assistant.io/open-mmlab/mmdetection3d?pullRequest=1280) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/open-mmlab/mmdetection3d?pullRequest=1280) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/open-mmlab/mmdetection3d?pullRequest=1280) it.

have agreed the Agreement

**Comment by hongye-dev** [comment]
Hello, thanks for your attention and reply .
well, I will prepare the  error log materails when i have some free time . 
yes, it is used for non-batch cases, seems not used frequently, so this problem may be not very urgent













At 2022-03-02 17:27:43, "Wenwei Zhang" ***@***.***> wrote:

Hi @hongye-dev ,
Could you post your error mesage or full error log in this PR? We should not modify that line as it is used for non-batch cases.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Tai-Wang** [comment]
@VVsssssk , please have a review here :)

File: mmdet3d/models/middle_encoders/pillar_scatter.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def forward_single(self, voxel_features, coors):
             dtype=voxel_features.dtype,
             device=voxel_features.device)
 
-        indices = coors[:, 1] * self.nx + coors[:, 2]
+        indices = coors[:, 2] * self.nx + coors[:, 3]
```
Comment (VVsssssk) [inline mmdet3d/models/middle_encoders/pillar_scatter.py:53]: Hi, you can `return canvas`  in forward_single directly to make it consistent with forward_batch.

**Pull request merged**